core/chains/evm/txm/storage/inmemory_store_test.go

```diff
@@ -0,0 +1,482 @@
+		ctx1, err := insertConfirmedTransaction(m, 0)
+		require.NoError(t, err)
+
+		ctx2, err := insertConfirmedTransaction(m, 1)
+		require.NoError(t, err)
+
+		ctxs, utxs, err := m.MarkTransactionsConfirmed(1)
+		require.NoError(t, err)
+		assert.Equal(t, types.TxConfirmed, ctx1.State)
+		assert.Equal(t, types.TxUnconfirmed, ctx2.State)
+		assert.Equal(t, utxs[0], ctx2.ID)
+		assert.Empty(t, ctxs)
+	})
+	t.Run("prunes confirmed transactions map if it reaches the limit", func(t *testing.T) {
+		m := NewInMemoryStore(logger.Test(t), fromAddress, testutils.FixtureChainID)
+		for i := 0; i < maxQueuedTransactions; i++ {
+			//nolint:gosec // this won't overflow
+			_, err := insertConfirmedTransaction(m, uint64(i))
+			require.NoError(t, err)
+		}
+		assert.Len(t, m.ConfirmedTransactions, maxQueuedTransactions)
+		_, _, err := m.MarkTransactionsConfirmed(maxQueuedTransactions)
+		require.NoError(t, err)
+		assert.Len(t, m.ConfirmedTransactions, (maxQueuedTransactions - maxQueuedTransactions/pruneSubset))
+	})
+}
+
+func TestMarkUnconfirmedTransactionPurgeable(t *testing.T) {
+	t.Parallel()
+
+	fromAddress := testutils.NewAddress()
+	m := NewInMemoryStore(logger.Test(t), fromAddress, testutils.FixtureChainID)
+
+	// fails if tx was not found
+	err := m.MarkUnconfirmedTransactionPurgeable(0)
+	require.Error(t, err)
+
+	tx, err := insertUnconfirmedTransaction(m, 0)
+	require.NoError(t, err)
+	err = m.MarkUnconfirmedTransactionPurgeable(0)
+	require.NoError(t, err)
+	assert.True(t, tx.IsPurgeable)
+}
+
+func TestUpdateTransactionBroadcast(t *testing.T) {
+	t.Parallel()
+
+	fromAddress := testutils.NewAddress()
+	hash := testutils.NewHash()
+	t.Run("fails if unconfirmed transaction was not found", func(t *testing.T) {
+		m := NewInMemoryStore(logger.Test(t), fromAddress, testutils.FixtureChainID)
+		var nonce uint64
+		require.Error(t, m.UpdateTransactionBroadcast(0, nonce, hash))
+	})
+
+	t.Run("fails if attempt was not found for a given transaction", func(t *testing.T) {
+		m := NewInMemoryStore(logger.Test(t), fromAddress, testutils.FixtureChainID)
+		var nonce uint64
+		tx, err := insertUnconfirmedTransaction(m, nonce)
+		require.NoError(t, err)
+		require.Error(t, m.UpdateTransactionBroadcast(0, nonce, hash))
+
+		// Attempt with different hash
+		attempt := &types.Attempt{TxID: tx.ID, Hash: testutils.NewHash()}
+		tx.Attempts = append(tx.Attempts, attempt)
+		require.Error(t, m.UpdateTransactionBroadcast(0, nonce, hash))
+	})
+
+	t.Run("updates transaction's and attempt's broadcast times", func(t *testing.T) {
+		m := NewInMemoryStore(logger.Test(t), fromAddress, testutils.FixtureChainID)
+		var nonce uint64
+		tx, err := insertUnconfirmedTransaction(m, nonce)
+		require.NoError(t, err)
+		attempt := &types.Attempt{TxID: tx.ID, Hash: hash}
+		tx.Attempts = append(tx.Attempts, attempt)
+		require.NoError(t, m.UpdateTransactionBroadcast(0, nonce, hash))
+		assert.False(t, tx.LastBroadcastAt.IsZero())
```

Comment:
Could you also check the tx `InitialBroadcastAt` too since this method sets that value too if 0.